Core/clim-core/frames.lisp

```diff
@@ -496,9 +496,13 @@ documentation produced by presentations.")
               (*partial-command-parser* partial-command-parser)
               (interactorp (typep *query-io* 'interactor-pane)))
          (restart-case
-             (progn
-               (redisplay-frame-panes frame :force-p first-time)
-               (setq first-time nil)
+             (labels ((exec-cmd (frame command)
+                        (when command
+                          (execute-frame-command frame command)
+                          (redisplay-frame-panes frame))))
```

Comment:
this is not right – command may change dynamic variable values and that would leak into redisplay.

Reply:
if that's fine I'll fix it on my own (a solution will be similar in spirit to what you did)